File: go-solutions/1493_longest_subarray.go
package leetcode

func LongestSubarray(nums []int) int {
	l := 0
	zeroCount := 0
	result := 0
	for r := 0; r < len(nums); r++ {
		if nums[r] == 0 {
			zeroCount++
		}
		for zeroCount > 1 {
			if nums[l] == 0 {
				zeroCount--
			}
			l++
		}
		result = max(result, r-l)
	}
	return result
}

File: ts-solutions/src/1493_longest_subarray.ts
export const longestSubarray = (nums: number[]) => {
	let left = 0
	let zeroes = 0
	let largestWindow = 0
	for (let i = 0; i < nums.length; i++) {
		if (nums[i] === 0) {
			zeroes++
		}
		while (zeroes > 1) {
			zeroes -= nums[left] === 0 ? 1 : 0
			left++
		}
		largestWindow = Math.max(largestWindow, i - left)
	}
	return largestWindow
}
